backend/internal/api/service_handlers.go

```diff
@@ -222,14 +223,56 @@ func (h *ServiceHandler) getViewportNodes(c *gin.Context) {
 
 // parseVault triggers a vault parsing operation
 func (h *ServiceHandler) parseVault(c *gin.Context) {
-	// TODO: Implement when VaultService is available
-	c.JSON(http.StatusNotImplemented, gin.H{"error": "Vault parsing not yet implemented"})
+	// Use request context
+	ctx := c.Request.Context()
+
+	// Check if vault service is available
+	if h.vaultService == nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": "Vault service not configured"})
+		return
+	}
+
+	// Check if parse is already running
+	status, err := h.vaultService.GetParseStatus(ctx)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": "Failed to get parse status"})
+		return
+	}
+
+	if status.Status == "parsing" {
+		c.JSON(http.StatusConflict, gin.H{"error": "Parse already in progress"})
+		return
+	}
+
+	// Start parsing
+	history, err := h.vaultService.ParseAndIndexVault(ctx)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
+		return
```

Comment:
TODO(CL): The ParseAndIndexVault call could take minutes for large vaults. Consider:
1. Making this endpoint async - return 202 Accepted immediately
2. Return a job ID for status tracking
3. Add webhook support for completion notifications

Example async response:
```json
{
  "job_id": "parse-12345",
  "status_url": "/api/v1/vault/parse/12345/status",
  "estimated_duration_seconds": 300
}
```